### ISSUE TITLE
SyntaxError: Unexpected token u when json.versions does not contain dist-tag

### ISSUE THREAD

**Issue by evilpacket** [davglass/registry-static#46]
When addVersionAs in lib/registry.js is called it makes the assumption that the passed in version from dist-tags is in json.versions. However there are situations in which it might not be.

```
 38     function addVersionAs(name, version) {
 39         parts.push({
 40             version: name,
 41             json: JSON.parse(JSON.stringify(json.versions[version]))
 42         });
 43     }
```

A module that demonstrates this behavior is [chokidar](https://npmjs.org/chokidar). The crash looks like this.

```
SyntaxError: Unexpected token u
    at Object.parse (native)
    at addVersionAs (/Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/registry.js:42:24)
    at /Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/registry.js:49:13
    at Array.forEach (native)
    at splitVersions (/Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/registry.js:47:40)
    at Object.split (/Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/registry.js:77:20)
    at /Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/index.js:23:29
    at Request._callback (/Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/follow-registry/lib/registry.js:26:9)
    at Request.self.callback (/Users/adam_baldwin/.nvm/versions/node/v4.2.1/lib/node_modules/registry-static/node_modules/request/request.js:198:22)
    at emitTwo (events.js:87:13)
```

registry-static -v 
2.0.0

node --version
v4.2.1

**Comment by krotscheck** [comment]
I can't reproduce this with latest, on a clean tree.

Examining the registry entry for chokidar, all of the `dist-tag` values appear to be available in `versions`:
http://registry.npmjs.org/chokidar

Furthermore, this works:
    `./bin/registry-static -d my.registry.com -o /tmp/registry/ --one chokidar`

Could you provide a bit more analysis of the problem you're encountering?

**Comment by evilpacket** [comment]
The rc6 entry does not appear to for me and is the one that it tanks on.

While that command you gave me works I do see this in the output

"015-10-26T17:11:39.223Z static [info] [undefined/undefined] writing json for chokidar to chokidar/index.json"

**Comment by bengl** [comment]
It's possible that the module was broken before. The code that breaks is here https://github.com/davglass/follow-registry/blob/master/lib/registry.js#L41 and should probably deal with bad data a little better.

**Comment by krotscheck** [comment]
Ah, gotit, my grep wasn't sophisticated enough. It seems like an easy enough problem to solve, but it'll have to be fixed in https://github.com/davglass/follow-registry.

The solution... well, the `versions` array that is constructed is really just a convenience data structure which is iterated over later, so I don't really see a problem adding a null check and just dropping any dist-tag for which the actual version can't be resolved.

**Comment by bengl** [comment]
I'm on it :)

**Comment by krotscheck** [comment]
Awesome!

**Comment by evilpacket** [comment]
<3 thanks!

**Comment by bengl** [comment]
No prob! Fix is in follow-registry@1.0.8, so an `npm install` of registry-static should pick it up.

**Comment by evilpacket** [comment]
works like a champ.
